### ISSUE TITLE
formatStyle using input data's value, e.g. color definition

### ISSUE THREAD

**Issue by Chr96er** [rstudio/DT#799]
I'm not sure whether this is intended behaviour but in years of using DT I haven't come across documentation for this.

What happened to me is: I have a data.table that (unexpectedly) has color definitions in a String column that serves as the value column for DT::formatStyle, as below (**Peru** corresponds to the color #cd853f, which I found out after a painful hour of debugging):

```
library(magrittr)
library(data.table)

dt_iris = iris %>% 
  data.table %>% 
  .[1:5] %>% 
  rbind(data.table(Species = "Peru"), fill = T) %>% 
  rbind(data.table(Species = "blue"), fill = T) %>% 
  rbind(data.table(Species = "#ABCDEF"), fill = T) %>% 
  rbind(data.table(Species = "foo"), fill = T)

dt_iris %>% 
  DT::datatable() %>% 
  DT::formatStyle(
    1,
    "Species",
    "row",
    background = styleEqual2(
      "setosa",
      "red"
    )
  )

```

Which produces the following:

![image](https://user-images.githubusercontent.com/8114107/79166315-2b8f0f00-7de5-11ea-8da2-63084d580039.png)

**update:** The generated html (which has `style="background: peru"` **but not** `style="background: foo"`) as well as the rowCallback which looks correct to me (`"function(row, data) {\nvar value=data[5]; $(row).css({'background':value == \"setosa\" ? \"red\" : value});\n}"`) make me think that there is some JS color recognition logic happening in the datatable package, so it's probably not a DT issue.

Unfortunately I don't know enough JS to be able to debug this.

```
xfun::session_info('DT')
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 19.10, RStudio 1.2.5033

Locale:
  LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8       
  LC_COLLATE=en_GB.UTF-8     LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
  LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
  LC_TELEPHONE=C             LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

Package version:
  assertthat_0.2.1   backports_1.1.4    BH_1.69.0.1        cli_1.1.0         
  colorspace_1.4.1   crayon_1.3.4       crosstalk_1.0.0    digest_0.6.21     
  DT_0.13.2          ellipsis_0.3.0     fansi_0.4.0        ggplot2_3.2.1     
  glue_1.3.1         graphics_3.6.1     grDevices_3.6.1    grid_3.6.1        
  gtable_0.3.0       htmltools_0.3.6    htmlwidgets_1.3    httpuv_1.5.2      
  jsonlite_1.6       labeling_0.3       later_0.8.0        lattice_0.20.38   
  lazyeval_0.2.2     magrittr_1.5       MASS_7.3.51.4      Matrix_1.2.17     
  methods_3.6.1      mgcv_1.8.28        mime_0.7           munsell_0.5.0     
  nlme_3.1.141       pillar_1.4.2       pkgconfig_2.0.3    plyr_1.8.4        
  promises_1.0.1     R6_2.4.0           RColorBrewer_1.1.2 Rcpp_1.0.2        
  reshape2_1.4.3     rlang_0.4.0        scales_1.0.0       shiny_1.3.2       
  sourcetools_0.1.7  splines_3.6.1      stats_3.6.1        stringi_1.4.3     
  stringr_1.4.0      tibble_2.1.3       tools_3.6.1        utf8_1.1.4        
  utils_3.6.1        vctrs_0.2.0        viridisLite_0.3.0  withr_2.1.2       
  xtable_1.8.4       yaml_2.2.0         zeallot_0.1.0  
```

**Comment by shrektan** [comment]
Thanks for the reporting. It looks like a bug to me. 

An unintended consequences of #702 (I don't know why I implemented it like that 😢 )

I will fix this ASAP.

**Comment by Mosk915** [comment]
I happened to come upon this and the behavior of this bug was actually something I've been trying to accomplish.  I have a column with the colors that I want to highlight each row.  I didn't know how to do it using `formatStyle` so I wrote a `rowCallback `to do it.  With this bug fixed, is there a way to still have `default `, or perhaps better yet `values` in the `styleEqual` function refer to a column rather than have to be a character vector?  If not, that would be a nice enhancement to have.
